src/concurrency.rs

```diff
@@ -41,13 +40,16 @@ impl Jobs {
     pub fn wait_for_all(&mut self) {
         while !self.jobs.is_empty() {
             let done: usize = {
-                let chans = self.jobs.iter().map(|j| &j.chan);
-                select! {
-                    recv(chans, msg, from) => {
-                        assert!(msg.is_none());
-                        self.jobs.iter().position(|j| &j.chan == from).unwrap()
-                    }
+                let mut select = Select::new();
+                for (index, chan) in self.jobs.iter().map(|j| &j.chan).enumerate() {
```

Comment:
Could we simplify this instead to:
```rust
for job in &self.jobs {
    select.recv(&job.chan);
}
```
?
I believe the assert here isn't really useful - this is as expected per [`Select::recv`](https://docs.rs/crossbeam-channel/0.3.6/crossbeam_channel/struct.Select.html#method.recv)'s docs and I think it's a lot more readable and concise in the short form.